list2/ex37/index.js

```diff
@@ -0,0 +1,40 @@
+import promptSync from "prompt-sync";
+const prompt = promptSync();
+
+const QUESTION_NUMBERS = 20;
+const STUDENT_NUMBERS = 50;
+const MEAN = 12;
+
+function examAnswers() {
+  const answerKey = [];
+  for (let i = 0; i < QUESTION_NUMBERS; i++) {
+    answerKey.push(Math.floor(Math.random() * 5) + 1);
+  }
+  return answerKey;
+}
+
+export function examScoreAnalysis() {
+  const answerKey = examAnswers();
+
+  for (let studentIndex = 0; studentIndex < STUDENT_NUMBERS; studentIndex++) {
+    const studentAnswerKey = [];
+    console.log(`Student (${studentIndex + 1}). Insert your answers, use just numbers (1-5).`);
+
+    for (let questionIndex = 0; questionIndex < QUESTION_NUMBERS; questionIndex++) {
+      console.log(`Question number ${questionIndex + 1}.`);
+      const studentAnswer = parseInt(prompt(">"));
+      studentAnswerKey.push(studentAnswer);
+    }
+    const correctQuestions = answerKey.filter((value) => studentAnswerKey.includes(value));
```

Comment:
This counts how many answers in the answerKey match any value in the student's answers but doesn't consider the position (question alignment). This logic is incorrect since it allows partial credit for matches in wrong positions. Please, make sure that you are accounting for the position.